DHFPROD-5048: Add steps to flow from Load TABLE view

briantang opened the pull request:
### Description

- Set default view of Load tile to 'Card' view
- Added functionality to add Load steps to a new or existing flow
- Added a few corresponding test cases
- Demo here:  https://drive.google.com/file/d/1yMX2GgxaqHRHXY9x4vTcAvQsbK4wq4gO/view
#### Checklist: 
```diff
- Note: do not change the below
```

-  ##### Owner:

- [x] JIRA_ID included in all the commit messages
- [x] PR title is in the format JIRA_ID:Title
- [x] Rebase the branch with upstream
- [x] Squashed all commits into a single commit
- [x] Added Tests
  

- ##### Reviewer:

- [x] Reviewed Tests

bsrikan commented on marklogic-data-hub-central/ui/src/components/load/load-list.tsx:
```diff
@@ -133,6 +218,9 @@ const LoadList: React.FC<Props> = (props) => {
             key: 'actions',
             render: (text, row) => (
                 <span>
+                    <Dropdown data-testid = "dropdown" overlay={menu(row.name)} trigger={['hover']} disabled = {!props.canWriteFlow}>
```
Replace with ```data-testid={`${row.name}-dropdown`}```

bsrikan commented on marklogic-data-hub-central/ui/src/components/load/load-list.tsx:
```diff
@@ -133,6 +218,9 @@ const LoadList: React.FC<Props> = (props) => {
             render: (text, row) => (
                 <span>
+                    <Dropdown data-testid = "dropdown" overlay={menu(row.name)} trigger={['hover']} disabled = {!props.canWriteFlow}>
+                        {props.canWriteFlow ? <span className={'AddToFlowIcon'} aria-label = "add-icon"></span> : <MLTooltip title={'Add to Flow'} placement="bottom"><span aria-label = "disabled-add-icon" className={'disabledAddToFlowIcon'}></span></MLTooltip>}
```
Replace aria-label(s) for add-icon and disabled-add-icon to include row.name

bsrikan commented on marklogic-data-hub-central/ui/src/pages/Load.tsx:
```diff
@@ -10,7 +10,7 @@ import { AuthoritiesContext } from "../util/authorities";
 export type ViewType =  'card' | 'list';
 
-const INITIAL_VIEW: ViewType = 'list';
+const INITIAL_VIEW: ViewType = 'card';
```
Can you pls verify Load.test if it needs some changes. It sets default view to be list and asserts the same.

briantang commented:
@bsrikan Thanks for catching some things I missed.

- Refactored your outlined testid's and aria-labels

- Added additional tests for 'Add step to a new flow' and 'Add step to an existing flow'

- Changed 'Add step to an existing flow' text to grey.

- Refactored Load.test tests to match new default Card view

wooldridge commented on marklogic-data-hub-central/ui/src/components/load/load-list.test.tsx:
```diff
@@ -109,4 +119,152 @@ describe('Load data component', () => {
+    fireEvent.click(getByText('Yes'));
+    expect(mockAddStepToFlow).toBeCalledTimes(1);
+
+    //TODO: Mock addStepToNew not implemented yet
```
Is the "add to new" testable now? Looks like we do something similar already in load-card.test.

briantang replied:
@wooldridge Right, was referring to the load-card.test where it is also still a 'TODO'. Seems like while the POST for addStepToFlow is supported and testable, the POST for addStepToNew is not implemented yet for us to test. Instead, adding a step to a new flow is done with a Link to /tiles/run/add. Not 100% sure but I think it makes sense, can refer to here as well. 

https://github.com/marklogic/marklogic-data-hub/blob/aea4246752d809fdce709317d841a249509a7d8f/marklogic-data-hub-central/ui/src/pages/Load.tsx#L112

wooldridge replied:
Thanks!
